src/qmapshack/gis/rte/router/brouter/CRouterBRouterDownloadPage.cpp

```diff
@@ -0,0 +1,256 @@
+/**********************************************************************************************
+    Copyright (C) 2023 Norbert Truchsess <norbert.truchsess@t-online.de>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+**********************************************************************************************/
+
+#include "gis/rte/router/brouter/CRouterBRouterDownloadPage.h"
+#include "gis/rte/router/brouter/CRouterBRouterSetup.h"
+
+#include <JlCompress.h>
+#include <QCheckBox>
+#include <QDir>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QMessageBox>
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+#include <QPushButton>
+
+CRouterBRouterReleaseItem::CRouterBRouterReleaseItem(const QString &_name, const QString &_description, const QString &_url) : name(_name), description(_description), url(_url) {}
+
+CRouterBRouterDownloadPage::CRouterBRouterDownloadPage() : QWizardPage() {
+  networkAccessManager = new QNetworkAccessManager(this);
+  networkAccessManager->setRedirectPolicy(QNetworkRequest::NoLessSafeRedirectPolicy);
+  connect(networkAccessManager, &QNetworkAccessManager::finished, this,
+          &CRouterBRouterDownloadPage::slotNetworkAccessManagerReplyReceived);
+  releaseItems = new QList<CRouterBRouterReleaseItem>();
```

Comment:
memory leak! `releaseItems` is never destroyed. Why do you allocate it on the heap in the first place?

Reply:
fixed